draft-ietf-idr-5g-edge-service-metadata-24.xml

```diff
@@ -435,33 +477,52 @@
+        <t>The Site Preference Index Sub-TLV has the following format:</t>
 
-        <figure anchor="Preference Index Sub-TLV"
-                title="Preference Index Sub-TLV">
+        <figure anchor="Site Preference Index Sub-TLV"
+                title="Site Preference Index Sub-TLV">
           <artwork><![CDATA[ 
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |Site-Preference-Index Sub-Type | Length        | Reserved      |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                   Preference Index value                      |
+  |                Site Preference Index value                    |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
 	]]></artwork>
         </figure>
+	<!-- XXX JMH - The normative text below should indicate the field
+	lengths.  PDU diagrams disagreeing with the normative text is
+	unfortunately common in IETF and can lead to interop issues.
+
+	If the sub-tlv format is described above as discussed in a prior
+	comment, there is no need to re-state the sub-type and length sizes -
+	but also no harm in doing so. -->
 
         <t><list style="hanging">
             <t hangText="-">Site-Preference-Index Sub-Type =1 (specified in
             this document).</t>
+	    <!-- XXX JMH - style comment: This is a sub-tlv of value 1, which is
+	    assigned for site-preference index.  It is not a a site-preference
+	    index sub-tlv.  Consider how you would like to name such things.
+	    -->
 
             <t hangText="-">Length: Specifies the total length in octets of
             the value field (not including the Type and Length fields). The
             Length = 5 for the Site-Preference-Index Sub-Type.</t>
+	    <!-- XXX JMH for these fixed-length fields, I'd suggest just saying
+	    "the length is 5". -->
 
             <t hangText="-">Reserved: Reserved for future use.</t>
+	    <!-- XXX JMH - The document needs to discuss what goes in reserved fields.  For example, "In this version of the document, the Reserved field MUST be set to zero and MUST be ignored upon receipt.  Received values MUST be propagated without change." -->
 
-            <t hangText="-">Preference Index value: 1 .. (2^32-1); the higher
-            the value, the more preference the site. Preference Index value ==
+            <t hangText="-">Site Preference Index value: 1 .. (2^32-1); the higher
+            the value, the more preference for the site. Site Preference Index value ==
             0 is reserved.</t>
+	    <!-- XXX JMH - Somewhere in the document you'll need to define what
+	    happens if you receive the reserved value of zero. -->
+	    <!-- Somewhere in the document you need to define what the byte
+	    order is; i.e. network byte order.  A good place to have done this
+	    is in the definition of sub-tlvs, or in each value. -->
```

Comment:
How about adding the following sentence to the end of Section 4.1 (Metadata Path Attribute)? 

"_The fields within the Metadata Path Attribute and its Sub-TLVs MUST use network byte order (big-endian), where the most significant byte is transmitted first_."

Reply:
In next push.